addons/project/models/project.py

```diff
@@ -2219,7 +2219,28 @@ def _get_default_color(self):
 
     name = fields.Char('Name', required=True, translate=True)
     color = fields.Integer(string='Color', default=_get_default_color)
+    project_ids = fields.Many2many('project.project', 'project_project_project_tags_rel', string='Projects')
+    task_ids = fields.Many2many('project.task', string='Tasks')
 
     _sql_constraints = [
         ('name_uniq', 'unique (name)', "Tag name already exists!"),
     ]
+
+    @api.model
+    def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get_uid=None):
+        domain = args
+        if 'project_id' in self.env.context:
+            project_id = self.env.context.get('project_id')
+            task_ids = self.env['project.task'].search([('project_id', '=', project_id)]).ids
+            domain = AND([
+                domain,
+                ['|', ('task_ids', 'in', task_ids), ('project_ids', 'in', project_id)]
+            ])
```

Comment:
I think that this could potentially crash in production as the number of unarchived task can be quite huge.
We should rather go get the tag_ids that are used in those task and add a 'id' in with those ids (as we expect to have less tag_ids than task_ids). You can thus remove the task_ids field.